pkg/receive/handler.go

```diff
@@ -377,25 +385,27 @@ func (h *Handler) parallelizeRequests(ctx context.Context, tenant string, replic
 				h.forwardRequestsTotal.WithLabelValues("success").Inc()
 			}()
 
+			cl, err := h.peers.Get(ctx, endpoint)
+			if err != nil {
+				level.Error(h.logger).Log("msg", "failed to get peer connection to forward request", "err", err, "endpoint", endpoint)
+				ec <- err
+				return
+			}
+
 			// Create a span to track the request made to another receive node.
 			span, ctx := tracing.StartSpan(ctx, "thanos_receive_forward")
 			defer span.Finish()
 
-			// Actually make the request against the endpoint
-			// we determined should handle these time series.
-			var res *http.Response
-			res, err = h.client.Do(req.WithContext(ctx))
+			_, err = cl.RemoteWrite(ctx, &storepb.WriteRequest{
```

Comment:
Let's keep this comment here

Reply:
I think this is missing the first part of the comment

Reply:
Fixed